tests/vnd/test_error.py

```diff
@@ -0,0 +1,69 @@
+# coding=utf8
+"""Test vnd.error."""
+
+
+import uuid
+
+import halogen
+from halogen.vnd.error import Error, VNDError
+
+
+class APIError(Error):
+
+    def __init__(self, status_code, **kwargs):
+        super(APIError, self).__init__(**kwargs)
+        self.status_code = status_code
+        self.logref = uuid.uuid4().hex
+
+
+class AuthorSchema(halogen.Schema):
+
+    name = halogen.Attr(required=True)
+
+
+class PublisherSchema(halogen.Schema):
+
+    name = halogen.Attr(required=True)
+    address = halogen.Attr()
+
+
+class BookSchema(halogen.Schema):
+
+    title = halogen.Attr(required=True)
+    year = halogen.Attr(halogen.types.Int(), required=True)
+    authors = halogen.Attr(halogen.types.List(AuthorSchema), required=True)
+    publisher = halogen.Attr(PublisherSchema)
+
+
+def test_validation():
+    try:
+        BookSchema.deserialize(
+            dict(
+                # title is skipped
+                year=u"☃bc",  # Not an integer
+                authors=[dict(name="John Smith"), dict()],  # Second author has no name
+                publisher=dict(address="Chasey Lane 42, Los Angeles, US"),  # No name
+            ),
+        )
+    except halogen.exceptions.ValidationError as e:
+        error = APIError.from_validation_exception(e, status_code=400)
+
+    assert error.status_code == 400
+
+    data = VNDError.serialize(error)
+    assert uuid.UUID(data["logref"])
+    assert data["message"] == "Validation error."
+
+    expected_errors = [
+        dict(path="/authors/1/name", message="Missing attribute."),
+        dict(path="/title", message="Missing attribute."),
+        dict(path="/year", message=u"'☃bc' is not an integer"),
+        dict(path="/publisher/name", message="Missing attribute."),
+    ]
+    expected = {e["path"]: e for e in expected_errors}
```

Comment:
can we do a manual check on data['_embedded']['errors'] instead? Without for-loops, just check thinks like
```python
errors = data['_embedded']['errors']
assert errors == [
    {'path': '/authors/1/name', 'message': 'Missing attribute.'},
    {'path': '/title', 'message': 'Missing attribute.'},
    ...,
]
```
This should work since ordering is guaranteed, and it makes it easier to see what we are testing and even how the functionality should work

Reply:
The output of the serialisation is OrderedDicts.. They are not compared that easily. This is just a version of the code that may change later.

Reply:
done